packages/launcher/src/index.ts

```diff
@@ -99,26 +102,69 @@ interface ILauncher {
 export
 interface ILauncherItem {
   /**
-   * The display name of the launcher item.
+   * The display name for the launcher item.
    */
-  name: string;
+  displayName: string;
 
   /**
-   * The ID of the command that is called to launch the item.
+   * The callback invoked to launch the item.
+   *
+   * The callback is invoked with a current working directory and the
+   * name of the selected launcher item.  When the function returns
+   * the launcher will close.
    */
-  command: string;
+  callback: (cwd: string, name: string) => Widget | Promise<Widget>;
 
   /**
-   * The command arguments, if any, needed to launch the item.
+   * The icon class for the launcher item.
+   *
+   * #### Notes
+   * This class name will be added to the icon node for the visual
+   * representation of the launcher item.
+   *
+   * Multiple class names can be separated with white space.
+   *
+   * The default value is an empty string.
    */
-  args?: JSONObject;
+  iconClass?: string;
 
   /**
-   * The image class name to attach to the launcher item. Defaults to
-   * 'jp-Image' followed by the `name` with spaces removed. So if the name is
-   * 'Launch New Terminal' the class name will be 'jp-ImageLaunchNewTerminal'.
+   * The icon label for the launcher item.
+   *
+   * #### Notes
+   * This label will be added as text to the icon node for the visual
+   * representation of the launcher item.
+   *
+   * The default value is an empty string.
    */
-  imgClassName?: string;
+  iconLabel?: string;
+
+  /**
+   * The identifier for the launcher item.
+   *
+   * The default value is the displayName.
+   */
+  name?: string;
+
+  /**
+   * The category for the launcher item.
+   *
+   * The default value is the an empty string.
+   */
+  category?: string;
```

Comment:
The only think I can imagine adding here later is a field for the kernel name. That could let us sort by category or kernel later. But we can always add that later.